cmd/infracost/run.go

```diff
@@ -103,6 +104,13 @@ func runMain(cmd *cobra.Command, runCtx *config.RunContext) error {
 			if err != nil {
 				return err
 			}
+			// TODO: reload resources to use freshly-synced usage estimates from cloud
+			//   - how to do this without drastically increasing runtime?
+			//   - we just need to recompute costs...maybe separate that from loading?
+			// err = provider.LoadResources(project, u)
```

Comment:
I'm ok with reloading the resources.  I'm expecting --sync-usage to be run fairly infrequently.  Also I'm curious how much time hitting the aws APIs will add.  It might be neat to add a 'runTime' metric to our mixpanel data.

Reply:
Cmurrently `LoadResources` does the following
* 1 - Runs Terraform init, plan, show and loads the resources into a `map[string]*schema.ResourceData`.
* 2 - Loops through the `map[string]*schema.ResourceData`, finds the matching usage data and constructs an array of `[]*schema.Resource`

My concern is that `#1` is often slow, since `terraform plan` hits the cloud vendor APIs. On larger projects I've seen this step take >3 minutes, so having to repeat this part might be an issue. I wonder if we could split this into those two stages and sync the usage data in between `#1` and `#2`, so we can pass the usage data when the resource is constructed?

Some ideas:
1. Refactor `syncResourcesUsage` to take `[]*schema.ResourceData` instead of `[]*schema.Resource`. Not looked into this too much yet, so might be difficult.
2. Somehow cache the terraform plan JSON when loading resources, so the second time it just loads it from the plan JSON.

Reply:
I solved this by caching plan + state JSON at the provider level, per @aliscott's suggestion.